File: FizzBuzz/Scala/src/main/scala/PrinterJob.scala
object PrinterJob {

  def print(result: Any): Any ={
    val printers = PrintersProvider.all()
    printers match {
      case Nil =>
        result
      case _ =>
        printWith(result, printers)
    }
  }

  private def printWith(value: Any, printers: List[(Any) => Any]): Any = {
    printers.head(value)
  }
}

File: FizzBuzz/java/src/main/java/PrinterJob.java
import java.util.List;

enum PrinterJob {
  $;

  static String print(final List<String> list) {
      final List<Printer> printers = PrintersProvider.all();
      if (!printers.isEmpty()) {
	  final Printer firstPrinter = printers.get(0);
	  if(firstPrinter != null)
	      {
		  return firstPrinter.print(list);
	      }
      }
      return DefaultPrinter.getInstance().print(list);
  }
}
